tools/src/download.c

```diff
@@ -2719,20 +2719,30 @@ int setRequestHeaders( HWTH_RETURNCODE_TYPE *rcPtr,
 char **getRequestHeaders() {
  char buf[80];
  char **headers = NULL;
- int NUM_HEADERS = 1;
+ int NUM_HEADERS = 2;
+
+ char *token = getenv("ZOPEN_GIT_OAUTH_TOKEN");
+ if (token == NULL) {
+	token = "";
+ }
 
 /***************************************************
  * Allocate an array of pointers to string, and
  * build a string for each of the headers.  Use a
  * NULL last array element to indicate end of array.
  ***************************************************/
  headers = (char **)calloc( 1+NUM_HEADERS, sizeof(char *) );
```

Comment:
malloc is good enough right? can you error out if the memory request fails (yes - i know i should've done that but since you're playing here maybe it's a good time to enhance it?

Reply:
I've added a check for the `calloc()` call.